### DISCUSSION TITLE
Could not find the drag and drop manager in the context of RootDropTargetsComponent

### DISCUSSION THREAD

**Discussion by zqqqqz2000** [nomcopter/react-mosaic#174] [Q&A]
when I install mosaic using `npm i react-mosaic-component`, npm shows a lot of warnings about resolve dependency(include dnd). then I run the demo, error below:
Could not find the drag and drop manager in the context of RootDropTargetsComponent. Make sure to render a DndProvider component in your top-level component. Read more: http://react-dnd.github.io/react-dnd/docs/troubleshooting#could-not-find-the-drag-and-drop-manager-in-the-context
![Screen Shot 2021-11-20 at 2 05 50 PM](https://user-images.githubusercontent.com/31151349/142716440-20123c39-1150-4ccd-a441-042c9b7cb650.png)

**Comment by Wozniaxos** [comment]
Had similiar issue, the reason why it works with yarn and not npm currently is that yarn by default won't install peerDependencies but npm 7+ does. It seems like some peerDeps are breaking the setup - haven't chance time to inspect what exactly causes the problem but when installed with --legacy-per-deeps flag which skips peerDependencies, everything works.

**Comment by nomcopter** [comment]
This is now fixed in [v5.2.2](https://github.com/nomcopter/react-mosaic/releases/tag/v5.2.2)
